sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -308,9 +308,11 @@ private ClientLogger getHttpLoggingPolicyLogger() {
      * @param url URL where the request is being sent.
      * @return A URL with query parameters redacted based on configurations in this policy.
      */
-    private static String getRedactedUrl(URL url, Set<String> allowedQueryParameterNames) {
-        return UrlBuilder.parse(url)
-            .setQuery(getAllowedQueryString(url.getQuery(), allowedQueryParameterNames))
+    private String getRedactedUrl(URL url, Set<String> allowedQueryParameterNames) {
+        UrlBuilder builder = UrlBuilder.parse(url);
+        String allowedQueryString = getAllowedQueryString(url.getQuery(), allowedQueryParameterNames);
+        builder.clearQuery();
```

Comment:
Is the `clearQuery` here is meant to remove any previously configured `Map` values? Would it be better to add another implementation method that handles parsing the `url.getQuery` into a `Map` which then replaces the uncleansed query values?

Reply:
@v-ddaian @alzimmermsft 

As I understood the code, the problem here is that `UrlBuilder.setQuery()` kind of lost its original meaning: previously it would call` with()` and with() would call `setQueryValue(String, String)` to build the query string. With the change in `with()` to call `appendQueryParameter()` instead, `name=SensitiveValue` is no longer be replaced by `name=REDACTED`. Rather, the result is `name=SensitiveValue&name=REDACTED`. Hence we now have the need to clear the query map before setQuery() is called, otherwise this appending behavior will prevent redaction from working. 

This is the problem, but is also required behavior for multiple query parameters with the same name to work.

In the current, I see that `clearQuery()` simply clears the query map, which works, provided that is called before `UrlBuilder.setQuery()` in `getRedactedURL()`. 

To me, this works well, but I would move clearing code inside setQuery(), for this method to retain the original semantics: "setQuery() replaces the query string in its entirety". I've tested it and it works. Would you see any problems with that approach, @alzimmermsft ? Doing so would also remove the current "2-phased" approach for a client: if you want the query set, you need to clear it first, yourself and would maintain the current implementation. 

Please let me know if you're ok with that approach. If not, can you please clarify a bit more what you are proposing? It seemed to me that you'd parse `url.getQuery()` into a map then use a new method in Builder to replace it? 

This is the last pending issue for us to close this PR, thanks for the support.

Reply:
Pinging @v-ddaian for opinion on the solution of moving builder.clearQuery() inside setQuery(), given he has made this change in the past.

Reply:
Hi, sounds good to me too.

Reply:
Let's keep this as-is given the potential implications, especially given this is just implementation that could be re-worked later (if needed) and maintain current functionality.